RecipeDetails/Views/RecipeDetailsViewController.swift

```diff
@@ -0,0 +1,295 @@
+            titleDescriptionLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 18),
+            titleDescriptionLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -18),
+            
+            recipeDescriptionLabel.topAnchor.constraint(equalTo: titleDescriptionLabel.bottomAnchor, constant: 10),
+            recipeDescriptionLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 18),
+            recipeDescriptionLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -18),
+            
+            nutrientsCollectionView.topAnchor.constraint(equalTo: recipeDescriptionLabel.bottomAnchor),
+            nutrientsCollectionView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            nutrientsCollectionView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            nutrientsCollectionView.bottomAnchor.constraint(equalTo: titleIngredientsLabel.topAnchor),
+            nutrientsCollectionView.heightAnchor.constraint(equalToConstant: 112),
+            
+            titleIngredientsLabel.topAnchor.constraint(equalTo: nutrientsCollectionView.bottomAnchor),
+            titleIngredientsLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 18),
+            titleIngredientsLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -18),
+            
+            ingredientsTableView.topAnchor.constraint(equalTo: titleIngredientsLabel.bottomAnchor, constant: 6),
+            ingredientsTableView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            ingredientsTableView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            
+            sourceLinkButton.topAnchor.constraint(equalTo: ingredientsTableView.bottomAnchor, constant: 18),
+            sourceLinkButton.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 18),
+            sourceLinkButton.heightAnchor.constraint(equalToConstant: 36),
+            sourceLinkButton.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -24),
+        ])
+    }
+}
+
+extension RecipeDetailsViewController: RecipeDetailsViewInput {
+    func configure(with data: Models.Recipe) {
+        title = data.label
+        recipeImageView.loadImage(for: data.images?.regular?.url)
+        
+        nutrientsCollectionViewDataSource = Array(data.digest?[0..<4] ?? [])
+        nutrientsCollectionView.reloadData()
+        ingredientsTableViewDataSource = data.ingredients ?? []
+        ingredientsTableView.reloadData()
+        
+        recipeDescriptionLabel.text = "Bring colour to your dinner table with our \(data.label ?? Texts.Discover.mockRecipeTitle). Packed with nutrients, it's a satisfying lunch or supper for the family"
+        
+        if UserDefaults.favouriteRecipes.contains(data) {
```

Comment:
Is it correct thing to do this here? It should be provided from the `interactor`

Reply:
Information about containing, I mean